src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -555,20 +558,36 @@ public void cleanupCourse(Long courseId) {
     public ResponseEntity<List<User>> getAllUsersInGroup(Course course, String groupName) {
         authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
         var usersInGroup = userRepository.findAllInGroup(groupName);
-        usersInGroup.forEach(user -> {
-            // remove some values which are not needed in the client
-            user.setLastNotificationRead(null);
-            user.setActivationKey(null);
-            user.setLangKey(null);
-            user.setLastNotificationRead(null);
-            user.setLastModifiedBy(null);
-            user.setLastModifiedDate(null);
-            user.setCreatedBy(null);
-            user.setCreatedDate(null);
-        });
+        removeUserVariables(usersInGroup);
         return ResponseEntity.ok().body(usersInGroup);
     }
 
+    /**
+     * Search for users of all user groups by login or name in course
+     *
+     * @param course        Course in which to search students
+     * @param nameOfUser    Login or name by which to search students
+     * @return users whose login matched
+     */
+    public List<User> searchOtherUsersNameInCourse(Course course, String nameOfUser) {
```

Comment:
"Name" was previously used by Stephan. Please take a look at `searchByLoginOrNameInGroup` method in `UserRepository.java`.

I also believe it should not really be ...Usernames... since we are concatenating first and last names of users to perform the search here. For this I think "name" as a more general term fits well however Users will also do. To conclude, I am changing the method's name to ...Users...